vendor/k8s.io/kube-openapi/pkg/handler/handler.go

```diff
@@ -202,3 +211,53 @@ func toGzip(data []byte) []byte {
 	zw.Close()
 	return buf.Bytes()
 }
+
+// RegisterOpenAPIVersionedService registers a handler to provides access to provided swagger spec.
+func RegisterOpenAPIVersionedService(openapiSpec *spec.Swagger, servePath string, handler common.PathHandler) (*OpenAPIService, error) {
+	o := OpenAPIService{}
+	if err := o.UpdateSpec(openapiSpec); err != nil {
+		return nil, err
+	}
+
+	files := map[string]func() ([]byte, string, time.Time){
+		"application/json":                                           o.getSwaggerBytes,
+		"application/com.github.proto-openapi.spec.v2@v1.0+protobuf": o.getSwaggerPbBytes,
+	}
+
+	handler.Handle(servePath, gziphandler.GzipHandler(http.HandlerFunc(
```

Comment:
Does this handle varying on Accept-Encoding as well?

Reply:
Yes, according to https://github.com/kubernetes/kubernetes/blob/f821a54d39dca103651a03b356de88c627cf720e/vendor/github.com/NYTimes/gziphandler/gzip.go#L245